src/bp/admin/ui/src/app/StatusBar/LangSwitcher.tsx

```diff
@@ -0,0 +1,81 @@
+import { Popover, Button, Position, PopoverInteractionKind, Menu, Colors } from '@blueprintjs/core'
+import React, { useEffect, FC } from 'react'
+import { connect, ConnectedProps } from 'react-redux'
+
+import { AppState } from '../rootReducer'
+import { setContentLang } from '../uiReducer'
+import style from './style.scss'
+
+const requireFlag = code => {
+  try {
+    const flag = require(`../../management/languages/flags/${code}.svg`)
+    return flag.default || flag
+  } catch (err) {
+    if (code === 'missing') {
+      return null
+    }
+
+    return requireFlag('missing')
+  }
+}
+
+const STORAGE_KEY = `bp::${window.BOT_ID}::cmsLanguage`
+
+type Props = ConnectedProps<typeof connector> & {
+  languages: string[]
+}
+
+const LangSwitcher: FC<Props> = props => {
+  useEffect(() => {
+    restoreLastLanguage()
+  }, [props.languages])
+
+  const restoreLastLanguage = () => {
+    const lastLang = localStorage.getItem(STORAGE_KEY)
+    if (!props.languages || !props.languages.length || !lastLang) {
```

Comment:
isn't optional chaining available through the admin?

Reply:
It does. Just did a lot of copy and paste without reviewing stuff